java/arcs/core/storage/handle/HandleManager.kt

```diff
@@ -0,0 +1,133 @@
+ *
+ * Handles that are used by end-users will deal with [RawEntity], so this helper only bothers to
+ * create those types.
+ *
+ * It will create a [StorageProxy] for each new storage key, and keep a reference to it for as
+ * long as the [HandleFactory] exists.
+ *
+ * Instantiate it with the context and lifecycle that should own the resulting activate stores.
+ */
+class HandleManager(private val aff: ActivationFactoryFactory? = null) {
+    companion object {
+        /**
+         * Convenience for making a ramdisk-backed reference mode key
+         */
+        fun ramdiskStorageKeyForName(name: String) = ReferenceModeStorageKey(
+            backingKey = RamDiskStorageKey("$name-backing"),
+            storageKey = RamDiskStorageKey("$name-storage")
+        )
+    }
+
+    private val singletonProxies = mutableMapOf<StorageKey, SingletonProxy<RawEntity>>()
+    private val setProxies = mutableMapOf<StorageKey, SetProxy<RawEntity>>()
+
+    private val singletonStores = mutableMapOf<StorageKey, SingletonStore<RawEntity>>()
+    private val setStores = mutableMapOf<StorageKey, SetStore<RawEntity>>()
+
+    /**
+     * Create a new SingletonHandle backed by an Android [ServiceStore]
+     */
+    suspend fun singletonHandle(
+        storageKey: StorageKey,
+        schema: Schema,
+        callbacks: Callbacks<SingletonOp<RawEntity>>? = null
+    ): SingletonHandle<RawEntity> {
+        val storeOptions = SingletonStoreOptions<RawEntity>(
+            storageKey = storageKey,
+            existenceCriteria = ExistenceCriteria.MayExist,
+            type = SingletonType(EntityType(schema)),
+            mode = StorageMode.ReferenceMode
+        )
+
+        val store = singletonStores.getOrPut(storageKey) {
+            Store(storeOptions)
+        }
+
+        val activeStore = aff?.let {
+            store.activate(it.singletonFactory())
+        } ?: store.activate()
+
+        val storageProxy = singletonProxies.getOrPut(storageKey) {
+            SingletonProxy(activeStore, CrdtSingleton())
+        }
+
+        return SingletonHandle(storageKey.toKeyString(), storageProxy).also {
+            storageProxy.registerHandle(it)
+            it.callback = callbacks
+        }
+    }
+
+    /**
+     * Create a new [SetHandle] backed by an Android [ServiceStore]
+     */
+    suspend fun setHandle(
+        storageKey: StorageKey,
+        schema: Schema,
+        callbacks: Callbacks<SetOp<RawEntity>>? = null
+    ): SetHandle<RawEntity> {
+        val storeOptions = SetStoreOptions<RawEntity>(
+            storageKey = storageKey,
+            existenceCriteria = ExistenceCriteria.MayExist,
+            type = CollectionType(EntityType(schema)),
+            mode = StorageMode.ReferenceMode
+        )
+
+        val store = setStores.getOrPut(storageKey) {
+            Store(storeOptions)
+        }
+
+        val activeStore = aff?.let {
+            store.activate(it.setFactory())
+        } ?: store.activate()
```

Comment:
Same, I think you could roll the nullable-typed `aff?.setFactory()` into the call to `activate` without the `?.let` and elvis.